### ISSUE TITLE
Question about versioning

### ISSUE THREAD

**Issue by RafalSkolasinski** [jupyter/docker-stacks#674]
Hi,

[Documentation](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html#versioning) stays that images are versioned by git commit SHA that triggered the image build. 

I wonder if it is guaranteed that e.g. [jupyter/scipy-notebook:7c45ec67c8e7 ](https://github.com/jupyter/docker-stacks/blob/7c45ec67c8e764bde29d58a34055214fe839ff44/scipy-notebook/Dockerfile) is for sure build from ``jupyter/minimal-notebook:7c45ec67c8e7``?

In its [Dockerfile](https://github.com/jupyter/docker-stacks/blob/7c45ec67c8e764bde29d58a34055214fe839ff44/scipy-notebook/Dockerfile#L3) it does not indicate which version of ``jupyter/minimal-notebook`` shall it be based on.

Second question, if answer is yes, how is it achieved?

Best,
Rafal

**Comment by parente** [comment]
tl;dr: Right now, it's only guaranteed when there are not multiple merges to master and build failures on Docker Cloud very close in time.

Longer explanation:

Every image folder in this repo has a [`hooks/post_push`](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/hooks/post_push) script. Docker Cloud executes this script after pushing a built image with tag `latest`. The script tags that `latest` image with a portion of the git SHA, pushes that tag as well, and then triggers downstream builds configured in Docker Cloud. Those downstream builds do receive the same git commit SHA that triggered the upstream build, and so pull the same source. But, as you pointed out, the child builds use `FROM <image>:latest` in their Dockerfile definitions, which may or may not match the git SHA. More often than not, `latest` does point to the parent image just built, the one that triggered the child builds. (In fact, our use of a free account may limit us to one build at a time per repository, but I'm not sure.) The exception occurs when a child build fails and is retriggered manually sometime later after the parent image `latest` has changed due to another merge to master. In practice, this rarely happens: the PR traffic to this repo is low and we typically trigger rebuilds before continuing with additional merges.

It may be possible to guarantee the parent-child SHA match using an `ARG` tag in the Dockerfile and additional Docker Cloud hooks. I haven't looked into what it would take to make change and ensure both local and Travis CI builds continue to work as-is. I'd be happy to review a PR if someone is interested in working on it.

**Comment by parente** [comment]
This issue has inactive for some time now and so I'm going to close it. #823 has some on-going work to expose more information about the content of builds.
